site/docs/v1/tech/themes/tailwind.adoc

```diff
@@ -0,0 +1,223 @@
+---
+:page-liquid:
+
+== Overview
+
+Tailwind CSS is a utility-first CSS framework for rapidly building custom user interfaces. If you are already using Tailwind CSS in your application, you can integrate it into FusionAuth.
+
+== Prerequisites
+
+* Node.js 18 or later
+
+== Setting up FusionAuth
+
+Create a new theme by navigating to [breadcrumb]#Customization -> Themes# and then click the [uielement]#Add# button. Enter a [field]#Name# for the theme and click the [uielement]#Save# button.
+
+image::themes/tailwind/create-theme.png[Create a theme,width=1200]
+
+Create a new API key with permissions to modify the theme. You can do this by going to [breadcrumb]#Settings -> API Keys#. Then click the [uielement]#Add# button and select `GET` and `PATCH` permissions for `/api/theme/`.
+
+image::themes/tailwind/create-api-key.png[Create an API key,width=1200]
+
+== Installation
+
+To get started, we need to create a new node project with the FusionAuth CLI and Tailwind CSS installed:
+
+[source,bash]
+----
+mkdir fusionauth-tailwind
+cd fusionauth-tailwind
+npm init -y
+npm install @fusionauth/cli tailwindcss@3
+npx tailwindcss init
+----
+
+This will install the dependencies and create the `tailwind.config.js` file in the project root. You can customize the configuration as needed.
+
+Change the `tailwind.config.js` to handle the FreeMarker template files in the `tpl` directory:
+
+[source,js]
+----
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ['./tpl/*.ftl'],
+  theme: {
+    extend: {}
+  },
+  plugins: []
+}
+----
+
+Create a new stylesheet `input.css` in the root directory:
+
+[source,css]
+----
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+----
+
+This stylesheet will be used to import the Tailwind CSS base, components, and utilities. Additionally, you can add custom classes to this stylesheet as well.
+
+We can now download the theme from FusionAuth:
+
+[source,bash]
+----
+npx fusionauth theme:download <themeId> -k <apiKey> -h http://localhost:9011
+----
+
+Replace `<themeId>` and `<apiKey>` with the ID of the theme and API key you created in FusionAuth respectively.
+
+Currently, the theme is using the `fusionauth-style.css` hosted in the FusionAuth instance. This could conflict with our Tailwind CSS file. To fix this, we need to remove the following line from the `tpl/helpers.ftl` file - it should be around line number 85:
+
+[source,html]
+----
+  <link rel="stylesheet" href="/css/fusionauth-style.css?version=${version}"/>
+----
+
+Finally, we add two new scripts to the `package.json` file:
+
+[source,json]
+----
+"scripts": {
+  "watch:tailwind": "tailwindcss -i ./input.css -o ./tpl/stylesheet.css --watch",
+  "watch:theme": "fusionauth theme:watch <themeId> -k <apiKey> -h http://localhost:9011"
+},
+----
+
+At this point, we have prepared the project to build the Tailwind CSS file and upload the theme to FusionAuth. To apply the changes in the `helper.ftl` file, run the following command:
```

Comment:
Is there a command to build the `./tpl/stylesheet.css` without watching the `input.css` file, like if I'm doing this from CI/CD?

Reply:
Yes, it the same but without the --watch flag: `tailwindcss -i ./input.css -o ./tpl/stylesheet.css`